R/sass-dependencies.R

```diff
@@ -0,0 +1,58 @@
+#' Compile Sass to CSS using LibSass, then return an HTML Dependencies
+#'
+#' Calls `sass(input, ...)` and wraps the output into a [htmltools::htmlDependency()].
+#'
+#' @inheritParams sass
+#' @inheritParams htmltools::htmlDependency
+#' @param cache_key additional information to include in the `sass_hash()` call used to
+#' determine the output directory. This should include any information that could possibly
+#' influence the resulting CSS that isn't already captured by `input`. For example, if
+#' `input` contains something like `"@import sass_file.scss"` you may want to include the
+#' [file.mtime()] of `sass_file.scss` (or, perhaps, a [packageVersion()] if `sass_file.scss`
+#' is bundled with an R package).
+#' @param stylesheet Absolute file path(s) pointing to stylesheet(s) to include with the dependency
+#'
+#' @return returns a list of [htmltools::htmlDependency()] objects. The length
+#' of the list use 1 unless `input` contains [sass_layer()] object(s) that
+#' have `html_deps`.
+#'
+#' @export
+#' @seealso [sass()]
+#' @examples
+#' str(sass_html_dependencies(
+#'   input = list(list(color = "red"), "body {color: $color}"),
+#'   name = "foo", version = "1.0.0",
+#'   stylesheet = "my-style.css"
+#' ))
+sass_html_dependencies <- function(
+  input, name, version, stylesheet, cache_key = NULL,
```

Comment:
`cache_key` makes it sound like it's the entire cache key, instead of just additional info to throw into the cache key. Maybe `cache_key_extra`? (Definitely open to other suggestions)

Reply:
I like that...and now that I think of it, it should probably be a `sass::sass()` arg